File: MyBaseApplication/app/src/main/java/gleroy/com/mybaseapplication/data/remote/api/base/MyApi.java
package gleroy.com.mybaseapplication.data.remote.api.base;

import gleroy.com.mybaseapplication.data.remote.entity.MyRemoteObject;
import retrofit2.Call;
import retrofit2.http.GET;

public interface MyApi {

    /*----- GET requests -----*/

    @GET("/google.com")
    Call<MyRemoteObject> getSomethingFromApi();
}

File: MyBaseApplication/app/src/main/java/gleroy/com/mybaseapplication/data/remote/api/base/MyApi.kt
package gleroy.com.mybaseapplication.data.remote.api.base

import gleroy.com.mybaseapplication.data.remote.entity.MyRemoteObject
import retrofit2.Call
import retrofit2.http.GET

interface MyApi {

    /*----- GET requests -----*/

    @get:GET("/todos/1")
    val somethingFromApi: Call<MyRemoteObject>
}
